pkg/services/object/get/exec.go

```diff
@@ -99,9 +99,13 @@ func (exec execCtx) address() oid.Address {
 	return exec.prm.addr
 }
 
+// isChild checks if reading object is a parent of the given object.
+// Object without reference to the parent (only children with the parent header
+// have it) is automatically considered as child: this should be guaranteed by
+// upper level logic.
 func (exec execCtx) isChild(obj *objectSDK.Object) bool {
 	par := obj.Parent()
-	return par != nil && equalAddresses(exec.address(), object.AddressOf(par))
+	return par == nil || equalAddresses(exec.address(), object.AddressOf(par))
```

Comment:
What about objects which have no parent? Do we ever execute this check for them (small or virtual objects)?

Reply:
You mean objects which have no parent at all? If so, then answer is no, we never call `isChild` for them.